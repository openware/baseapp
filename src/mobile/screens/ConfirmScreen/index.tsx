import * as React from 'react';
import { useIntl } from 'react-intl';
import { useSelector } from 'react-redux';
import { selectUserInfo } from '../../../modules/user/profile';
import { ConfirmScreen } from '../../../screens/ConfirmScreen';
import { Modal } from '../../components/Modal';

const ConfirmMobileScreen = () => {
    const intl = useIntl();
    const user = useSelector(selectUserInfo) || { level: 1 };

    return (
        <div className="cr-mobile-confirm">
            {
                user.level === 1 ?
                    <div className="cr-mobile-confirm__phone">
                        <Modal
                            isOpen
                            title={intl.formatMessage({id: 'page.body.profile.header.account.profile.phone.unverified.title'})}>
                            <ConfirmScreen/>
                        </Modal>
                    </div>
                    :
                    <div className="cr-mobile-confirm__identity">
                        <ConfirmScreen/>
                    </div>
            }
        </div>
    );
};

export {
    ConfirmMobileScreen,
};
