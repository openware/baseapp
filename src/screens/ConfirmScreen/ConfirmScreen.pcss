.pg-confirm {
  align-items: center;
  background: var(--body-background-color);
  display: flex;
  flex-direction: column;
  height: max-content;
  justify-content: flex-start !important;
  margin: 0 !important;
  margin-top: calc(var(--gap) * -1.3) !important;
  min-height: 100vh;
  padding: 0 10px;
  width: 100% !important;

  &__logo {
    height: 27px;
    margin-bottom: 24px;
    margin-top: 47px;
  }

  &__title {
    align-items: center;
    color: var(--primary-text-color);
    display: flex;
    font-size: 18px;
    font-style: normal;
    font-weight: 500;
    letter-spacing: 0.25px;
    line-height: 30px;
    margin: 0;
    margin-block-end: 0;
    margin-block-start: 0;
    margin-bottom: 6px;
    margin-inline-end: 0;
    margin-inline-start: 0;
    text-align: center;
  }

  &__content {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    margin-bottom: 100px;

    &-phone,
    &-identity,
    &-documents,
    &-address {
      max-width: 380px;
      padding: 0 10px;

      &__row,
      &__forms__row {
        align-items: flex-start;
        display: flex;
        flex-direction: row;
        height: max-content;
        justify-content: space-between;
        width: 100%;

        &__content {
          width: 100%;

          &--focused {
            border: 1px solid var(--primary-cta-color) !important;
            color: var(--primary-cta-color) !important;

            label {
              color: var(--primary-cta-color) !important;
            }
          }

          &--wrong {
            border: 1px solid var(--asks) !important;
            color: var(--asks) !important;

            label {
              color: var(--asks) !important;
            }
          }

          &-label {
            background: var(--input-background-color);
            color: var(--secondary-contrast-cta-color);
            font-size: 11px;
            line-height: 15px;
            margin-left: 9px;
            margin-top: 17px;
            padding: 0 5px;
            position: absolute;
            z-index: 1;
          }
        }
      }

      &__row.input-group > fieldset,
      &__forms__row.input-group > fieldset {
        width: calc(50% - 6px);

        &:first-child {
          margin-right: 6px;
        }

        &:last-child {
          margin-left: 6px;
        }
      }

      fieldset {
        background: var(--body-background-color);
        border: 1px solid var(--input-border-color);
        border-radius: 4px;
        height: 49px;
        margin-top: 24px;
        padding: 0;
        position: relative;
        width: 100%;

        .input-group {
          align-items: center;
          display: flex;
          height: 100%;
          justify-content: space-between;
          width: 100%;

          .custom-input {
            width: calc(100% - 150px) !important;
          }
        }

        .custom-input {
          height: 100%;
          width: calc(100% - 13px) !important;

          label {
            background: var(--input-background-color);
            color: var(--secondary-contrast-cta-color);
            font-size: 11px;
            left: 8px;
            line-height: 15px;
            margin-top: -8px;
            padding: 0 5px;
          }

          input {
            border: none;
            border-radius: 4px;
            color: var(--primary-text-color);
            font-size: 14px;
            font-weight: 500;
            letter-spacing: 0.1px;
            line-height: 22px;
            outline: none;
            padding-left: 12px !important;
            width: 100%;

            &::placeholder {
              opacity: 0.3;
            }
          }
        }

        button {
          border: none;
          border-radius: 4px !important;
          font-size: 12px;
          font-weight: 500;
          height: 34px;
          letter-spacing: 1px;
          line-height: 14px;
          margin-right: 7px;
          outline: none;
          padding: 0;
          text-transform: uppercase;
          width: 136px;

          &:focus,
          &:hover {
            border: none !important;
          }
        }
      }

      .cr-dropdown--default {
        button {
          color: var(--input-placeholder-color);
        }
      }

      .dropdown {
        button {
          align-items: center;
          display: flex;
          flex-direction: row;
          font-size: 14px;
          font-weight: 500;
          height: 49px;
          justify-content: space-between;
          letter-spacing: 0.1px;
          margin-top: 24px;
          outline: none;
          overflow-x: auto;
          width: 100%;

          &::-webkit-scrollbar {
            height: 0;
          }

          &::after {
            margin-right: 5px;
          }
        }

        &-menu {
          max-height: 98vh;
          overflow-x: hidden;
          overflow-y: auto;
        }
      }
    }
  }

  &__content-deep {
    margin: 0;
    margin-top: 24px;
    max-width: 100%;
    width: 100vw;

    button {
      border: none;
      border-radius: 4px;
      font-size: 14px;
      font-weight: 500;
      height: 40px;
      letter-spacing: 1px;
      line-height: 16px;
      text-transform: uppercase;
      transition: 0.2s linear;
    }
  }
}
