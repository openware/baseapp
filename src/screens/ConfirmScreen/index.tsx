import classnames from 'classnames';
import { History } from 'history';
import * as React from 'react';
import { FormattedMessage, InjectedIntlProps, injectIntl } from 'react-intl';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Phone } from '../../containers';
import { Documents } from '../../containers/Confirm/Documents';
import { Identity } from '../../containers/Confirm/Identity';
import { setDocumentTitle } from '../../helpers';
import {
    Label,
    labelFetch,
    RootState,
    selectCurrentColorTheme,
    selectLabelData,
    selectUserInfo,
    User,
} from '../../modules';

import logo from '../../assets/images/logo.svg';
import logoLight from '../../assets/images/logoLight.svg';

interface ReduxProps {
    colorTheme: string;
    userData: User;
    labels: Label[];
}

interface HistoryProps {
    history: History;
}

interface ConfirmState {
    title: string;
    level: number;
}

interface DispatchProps {
    labelFetch: typeof labelFetch;
}

type Props = ReduxProps & HistoryProps & DispatchProps & InjectedIntlProps;

class ConfirmComponent extends React.Component<Props, ConfirmState> {
    constructor(props: Props) {
        super(props);

        this.state = {
            title: '',
            level: 1,
        };
    }

    public componentDidMount() {
        setDocumentTitle('Confirm');
        this.props.labelFetch();
        const { userData } = this.props;
        this.setState({
            level: userData.level,
        });
    }

    public componentWillReceiveProps(next: Props) {
        if (next.userData.level !== this.state.level) {
            this.setState({
                level: next.userData.level,
            });
        }
    }

    public goBack = event => {
      event.preventDefault();
      this.props.history.goBack();
    };

    // tslint:disable:jsx-no-multiline-js
    public render() {
        const {
            colorTheme,
            userData,
            labels,
        } = this.props;
        const isIdentity = labels.find(w => w.key === 'profile' && w.value === 'verified');
        const currentProfileLevel = userData.level;
        const cx = classnames('pg-confirm__progress-items', {
            'pg-confirm__progress-first': currentProfileLevel === 1,
            'pg-confirm__progress-second': currentProfileLevel === 2 && !isIdentity,
            'pg-confirm__progress-third': currentProfileLevel === 3 || isIdentity,
        });
        return (
            <div className="pg-wrapper">
                <div className="pg-logo">
                    {colorTheme === 'light' ? (
                        <img src={logoLight} className="pg-logo__img" alt="Logo" />
                    ) : (
                        <img src={logo} className="pg-logo__img" alt="Logo" />
                    )}
                </div>
                <div className="pg-confirm">
                    <div className="pg-confirm-box">
                        <div
                            onClick={this.goBack}
                            className="pg-confirm-box-close"
                        />
                        <div className="pg-confirm__progress">
                            <div className={cx}>
                                <div className="pg-confirm__progress-circle-1">
                                    <span className="pg-confirm__title-text pg-confirm__active-1">
                                    <FormattedMessage id="page.body.kyc.head.phone"/>
                                    </span>
                                </div>
                                <div className="pg-confirm__progress-line-1" />
                                <div className="pg-confirm__progress-circle-2">
                                    <span className="pg-confirm__title-text pg-confirm__active-2">
                                    <FormattedMessage id="page.body.kyc.head.identity"/>
                                    </span>
                                </div>
                                <div className="pg-confirm__progress-line-2" />
                                <div className="pg-confirm__progress-circle-3">
                                    <span className="pg-confirm__title-text pg-confirm__active-3">
                                    <FormattedMessage id="page.body.kyc.head.document"/>
                                    </span>
                                </div>
                            </div>
                        </div>
                        <div className="pg-confirm__content">
                            {this.renderContent()}
                        </div>
                    </div>
                </div>
            </div>
        );
    }
    //tslint:enable:jsx-no-multiline-js

    private renderContent = () => {
        const { labels } = this.props;
        const { level } = this.state;
        const isIdentity = labels.find(w => w.key === 'profile' && w.value === 'verified');

        switch (level) {
            case 1: return <Phone />;
            case 2: return isIdentity ? <Documents /> : <Identity />;
            case 3: return <Documents />;
            default: return 'Something went wrong';
        }
    };
}

const mapStateToProps = (state: RootState): ReduxProps => ({
    colorTheme: selectCurrentColorTheme(state),
    userData: selectUserInfo(state),
    labels: selectLabelData(state),
});

const mapDispatchToProps = dispatch => ({
    labelFetch: () => dispatch(labelFetch()),
});

// tslint:disable-next-line
export const ConfirmScreen = injectIntl(withRouter(connect(mapStateToProps, mapDispatchToProps)(ConfirmComponent) as any));
