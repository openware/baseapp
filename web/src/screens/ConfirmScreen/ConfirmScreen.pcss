.pg-confirm {
  align-items: center;
  background: var(--body-background-color);
  display: flex;
  flex-direction: column;
  height: max-content;
  justify-content: flex-start !important;
  margin: 0 !important;
  margin-top: calc(var(--gap) * -1.3) !important;
  min-height: 100vh;
  padding: 0 10px;
  width: 100% !important;

  &__logo {
    margin-bottom: 24px;
    margin-top: 47px;

    img {
      height: auto;
      max-height: 250px;
      max-width: 360px;
      object-fit: contain;
    }
  }

  &__title {
    align-items: center;
    color: var(--primary-text-color);
    display: flex;
    font-size: 18px;
    font-style: normal;
    font-weight: 500;
    justify-content: center;
    letter-spacing: 0.25px;
    line-height: 30px;
    margin: 0;
    margin-block-end: 0;
    margin-block-start: 0;
    margin-bottom: 6px;
    margin-inline-end: 0;
    margin-inline-start: 0;
    text-align: center;

    &__icon {
      cursor: pointer;
      height: 15px;
      margin-left: 10px;
      width: 15px;

      path {
        fill: var(--icons);
      }
    }
  }

  &__content {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    margin-bottom: 100px;

    &-phone,
    &-identity,
    &-documents,
    &-address {
      max-width: 380px;
      padding: 0 10px;

      &__row,
      &__forms__row {
        align-items: flex-start;
        display: flex;
        flex-direction: row;
        height: max-content;
        justify-content: space-between;
        width: 100%;

        &__content {
          position: relative;
          width: 100%;

          &--focused {
            background: var(--body-background-color) !important;
            border: 1px solid var(--primary-cta-color) !important;
            color: var(--primary-cta-color) !important;

            label {
              background: var(--body-background-color) !important;
              color: var(--primary-cta-color) !important;
            }

            .input-group {
              background: var(--body-background-color) !important;

              input {
                background: var(--body-background-color) !important;
              }
            }
          }

          &--wrong {
            border: 1px solid var(--asks) !important;
            color: var(--asks) !important;

            label {
              color: var(--asks) !important;
            }
          }

          &-label {
            background: var(--body-background-color);
            font-size: 10px;
            font-weight: bold;
            left: 0;
            letter-spacing: 0.06em;
            padding: 0;
            position: absolute;
            text-transform: uppercase;
          }

          &-number-dropdown {
            margin-top: 24px;

            > * {
              color: var(--text) !important;
              font-family: var(--font-family) !important;
              font-size: var(--font-size) !important;
            }
          }
        }
      }

      &__row.input-group > fieldset,
      &__forms__row.input-group > fieldset {
        width: calc(50% - 6px);

        &:first-child {
          margin-right: 6px;
        }

        &:last-child {
          margin-left: 6px;
        }
      }

      fieldset {
        background: var(--input-background-color);
        border: 1px solid var(--input-border-color);
        border-radius: 4px;
        margin-top: 24px;
        padding: 1px;
        position: relative;
        width: 100%;

        .input-group {
          align-items: center;
          background: var(--input-background-color);
          display: flex;
          height: 100%;
          justify-content: space-between;
          width: 100%;

          .custom-input {
            width: calc(100% - 150px) !important;
          }

          input:-webkit-autofill,
          input:-webkit-autofill:hover,
          input:-webkit-autofill:focus {
            border: none;
          }
        }

        .custom-input {
          background: var(--input-background-color);
          border-radius: 4px;

          input {
            border: none;
            border-radius: 4px;
            color: var(--primary-text-color);
            font-size: 14px;
            font-weight: 500;
            height: 48px !important;
            letter-spacing: 0.1px;
            line-height: 22px;
            outline: none;
            padding-left: 12px !important;
            width: 100%;

            &::placeholder {
              opacity: 0.3;
            }
          }

          label {
            background: var(--body-background-color);
            font-size: 10px;
            font-weight: bold;
            left: 0;
            letter-spacing: 0.06em;
            margin-top: -18px;
            padding: 0;
            position: absolute;
            text-transform: uppercase;
          }
        }

        button {
          border: none;
          border-radius: 4px !important;
          font-size: 12px;
          font-weight: 500;
          height: 34px;
          letter-spacing: 1px;
          line-height: 14px;
          margin-right: 7px;
          outline: none;
          padding: 0;
          text-transform: uppercase;
          width: 136px;

          &:focus,
          &:hover {
            border: none !important;
          }
        }
      }

      .cr-dropdown--default {
        button {
          color: var(--input-placeholder-color);
        }
      }

      .dropdown {
        button {
          align-items: center;
          display: flex;
          flex-direction: row;
          font-size: 14px;
          font-weight: 500;
          height: 49px;
          justify-content: space-between;
          letter-spacing: 0.1px;
          margin-top: 24px;
          outline: none;
          overflow-x: auto;
          width: 100%;

          &::-webkit-scrollbar {
            height: 0;
          }

          &::after {
            margin-right: 5px;
          }
        }

        &-menu {
          max-height: 98vh;
          overflow-x: hidden;
          overflow-y: auto;
        }
      }
    }

    &-address__row__content-label {
      margin-top: -18px;
    }
  }

  &__content-deep {
    margin: 0;
    margin-top: 24px;
    max-width: 100%;
    width: 100vw;

    button {
      align-items: center;
      border: none;
      border-radius: 4px;
      display: flex;
      font-size: 14px;
      font-weight: 500;
      height: 40px;
      justify-content: center;
      letter-spacing: 1px;
      line-height: 16px;
      text-transform: uppercase;
      transition: 0.2s linear;
    }
  }

  .info {
    margin-top: 18px;
    max-width: 360px;
  }
}
