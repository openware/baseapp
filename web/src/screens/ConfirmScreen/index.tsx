import * as React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import { connect } from 'react-redux';
import { RouterProps } from 'react-router';
import { withRouter } from 'react-router-dom';
import { compose } from 'redux';
import { IntlProps } from '../../';
import { kycSteps } from '../../api';
import { CrossIcon } from '../../assets/images/kyc/CrossIcon';
import { Info, Logo } from '../../components';
import { Address, Documents, Identity, Phone } from '../../containers';
import { getVerificationStep, setDocumentTitle } from '../../helpers';
import {
    Label,
    labelFetch,
    RootState,
    selectLabelData,
    selectSidebarState,
    toggleSidebar,
} from '../../modules';

interface ReduxProps {
    isSidebarOpen: boolean;
    labels: Label[];
}

interface DispatchProps {
    labelFetch: typeof labelFetch;
    toggleSidebar: typeof toggleSidebar;
}

type Props = ReduxProps & DispatchProps & RouterProps & IntlProps;

class ConfirmComponent extends React.Component<Props> {
    public componentDidMount() {
        const { labels, isSidebarOpen } = this.props;
        setDocumentTitle('Confirm');
        this.props.labelFetch();

        if (labels.length) {
            this.handleCheckUserLabels(labels);
        }

        isSidebarOpen && this.props.toggleSidebar(false);
    }

    public componentDidUpdate(prevProps: Props) {
        const { labels } = this.props;

        if (labels.length && JSON.stringify(labels) !== JSON.stringify(prevProps.labels)) {
            this.handleCheckUserLabels(labels);
        }
    }

    public renderVerificationStep = (step: string) => {
        switch (step) {
            case 'phone':    return <Phone />;
            case 'profile':  return <Identity />;
            case 'document': return <Documents />;
            case 'address':  return <Address />;
            default: return 'Something went wrong';
        }
    };

    public render() {
        const { history } = this.props;

        const step = this.handleGetVerificationStep();

        return (
            <div className="pg-container pg-confirm">
                <div className="pg-confirm__logo">
                    <Logo />
                </div>
                <h3 className="pg-confirm__title">
                    <FormattedMessage id={`page.confirm.title.${step}`} />
                    <CrossIcon
                        className="pg-confirm__title__icon"
                        onClick={e => history.push('/profile')}
                    />
                </h3>
                {['profile', 'address'].includes(step) && <Info variant="warning" text={this.translate('page.confirm.title.warning')} />}
                <div className="pg-confirm__content">
                    {this.renderVerificationStep(step)}
                </div>
            </div>
        );
    }

    public translate = (id: string) => this.props.intl.formatMessage({ id });

    private handleGetVerificationStep = (): string => {
        const { labels } = this.props;

        return getVerificationStep(labels);
    };

    private handleCheckUserLabels = (labels: Label[]) => {
        const pendingLabelExists = Boolean(labels.find(label => kycSteps().includes(label.key) && ['pending', 'drafted', 'submitted'].includes(label.value) && label.scope === 'private'));
        const passedSteps = kycSteps().filter((step: string) => labels.find(label => step === label.key && label.value === 'verified' && label.scope === 'private'));

        if (pendingLabelExists || (kycSteps().length === passedSteps.length)) {
            this.props.history.push('/profile');
        }
    };
}

const mapStateToProps = (state: RootState): ReduxProps => ({
    isSidebarOpen: selectSidebarState(state),
    labels: selectLabelData(state),
});

const mapDispatchToProps = dispatch => ({
    labelFetch: () => dispatch(labelFetch()),
    toggleSidebar: (payload) => dispatch(toggleSidebar(payload)),
});

export const ConfirmScreen = compose(
    injectIntl,
    withRouter,
    connect(mapStateToProps, mapDispatchToProps),
)(ConfirmComponent) as any; // tslint:disable-line
