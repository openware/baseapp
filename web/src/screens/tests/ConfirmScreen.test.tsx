import { shallow } from 'enzyme';
import { TestComponentWrapper } from 'lib/test';
import * as React from 'react';
import { IntlProps } from '../../';
import { ConfirmScreen } from '../ConfirmScreen';

const setup = (props: Partial<IntlProps> = {}) =>
    shallow(
        <TestComponentWrapper>
            <ConfirmScreen />
        </TestComponentWrapper>,
    );

describe('ConfirmScreen test', () => {
    it('should render', () => {
        const wrapper = setup().render();
        expect(wrapper).toMatchSnapshot();
    });
});
