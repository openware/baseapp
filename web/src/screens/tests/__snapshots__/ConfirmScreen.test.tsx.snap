// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmScreen test should render 1`] = `
<div
  class="pg-container pg-confirm"
>
  <div
    class="pg-confirm__logo"
  >
    <div
      class="pg-logo"
    >
      <svg
        class="pg-logo__img"
        fill="none"
        height="20"
        style="width:auto"
        viewBox="0 0 118 20"
        width="118"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0.384613 17.1605C0.384613 18.7911 1.19766 19.5924 2.8651 19.5924H8.21193C12.1945 19.5924 14.7577 17.4698 14.7577 14.1804C14.7577 11.5658 13.104 9.7384 10.5684 9.49943V9.33075C12.5803 8.85281 13.793 7.29248 13.793 5.1277C13.793 2.23196 11.6433 0.404545 8.22571 0.404545H2.8651C1.19766 0.404545 0.384613 1.21985 0.384613 2.83641V17.1605ZM4.07778 8.38893V3.25812H7.37132C9.10766 3.25812 10.1274 4.21399 10.1274 5.77432C10.1274 7.44711 9.02498 8.38893 6.53071 8.38893H4.07778ZM4.07778 16.7388V11.0316H7.60559C9.78291 11.0316 10.968 12.1421 10.968 13.9555C10.968 15.7829 9.76912 16.7388 6.83388 16.7388H4.07778Z"
          fill="var(--primary-text-color)"
        />
        <path
          d="M18.0925 19.761C19.0847 19.761 19.636 19.2409 20.0218 17.9196L21.0691 14.6584H27.67L28.7311 17.9196C29.1031 19.2269 29.6819 19.761 30.6879 19.761C31.749 19.761 32.4518 19.1144 32.4518 18.1023C32.4518 17.7087 32.3553 17.2448 32.1624 16.6404L27.4219 2.78018C26.8294 0.980882 26.0025 0.249918 24.4453 0.249918C22.8744 0.249918 22.0338 0.994939 21.4412 2.78018L16.7007 16.6404C16.4664 17.3573 16.37 17.7931 16.37 18.1726C16.37 19.1285 17.0728 19.761 18.0925 19.761ZM21.896 11.8329L24.2938 4.07342H24.4729L26.8569 11.8329H21.896Z"
          fill="var(--primary-text-color)"
        />
        <path
          d="M34.2019 15.6423C34.2019 16.4436 34.6705 17.3011 35.5386 18.1164C36.7789 19.3112 38.9424 20 41.3816 20C45.874 20 48.7266 17.6806 48.7266 14.0258C48.7266 11.2003 47.1005 9.49943 43.6553 8.69818L40.8303 8.00939C39.0802 7.60173 38.2258 6.75831 38.2258 5.56347C38.2258 4.07342 39.4799 3.11755 41.4367 3.11755C43.0628 3.11755 44.3306 3.83446 45.1712 5.11364C45.6122 5.67592 46.1083 5.92895 46.6871 5.92895C47.5828 5.92895 48.1891 5.3245 48.1891 4.43891C48.1891 3.55331 47.7344 2.65367 46.88 1.86647C45.6397 0.699742 43.6278 0.0390625 41.3816 0.0390625C37.1923 0.0390625 34.4638 2.33035 34.4638 5.85866C34.4638 8.64195 36.0761 10.3991 39.3145 11.1582L42.1257 11.861C44.1652 12.353 45.0472 13.1543 45.0472 14.4475C45.0472 16.0078 43.7932 16.9075 41.5883 16.9075C39.9484 16.9075 38.5428 16.2609 37.4817 15.0098C36.8202 14.2788 36.3103 14.0117 35.6902 14.0117C34.8496 14.0117 34.2019 14.6865 34.2019 15.6423Z"
          fill="var(--primary-text-color)"
        />
        <path
          d="M51.7583 17.1605C51.7583 18.7911 52.5713 19.5924 54.2388 19.5924H62.5208C63.4992 19.5924 64.1332 18.9879 64.1332 18.0461C64.1332 17.1043 63.4992 16.4858 62.5208 16.4858H55.4514V11.3409H62.1763C63.0583 11.3409 63.6508 10.7786 63.6508 9.89303C63.6508 9.00743 63.0583 8.4311 62.1763 8.4311H55.4514V3.51114H62.5622C63.4992 3.51114 64.1332 2.89264 64.1332 1.96487C64.1332 1.03711 63.4992 0.404545 62.5622 0.404545H54.2388C52.5713 0.404545 51.7583 1.21985 51.7583 2.83641V17.1605Z"
          fill="var(--primary-text-color)"
        />
        <path
          clip-rule="evenodd"
          d="M75.0182 0C73.5028 0 72.2744 1.25311 72.2744 2.79891V17.1933C72.2744 18.7391 73.5028 19.9922 75.0182 19.9922H114.314C115.829 19.9922 117.058 18.7391 117.058 17.1933V2.79891C117.058 1.25311 115.829 0 114.314 0H75.0182ZM84.8718 14.3366H84.7998C84.1516 15.407 83.0509 15.9841 81.6826 15.9841C79.5737 15.9841 78.154 14.6199 78.154 12.5946C78.154 10.5902 79.584 9.34147 81.9604 9.34147H84.8203V8.27109C84.8203 6.8544 83.987 6.13032 82.4336 6.13032C81.518 6.13032 80.9008 6.45563 80.2527 7.2112C80.0058 7.50503 79.7692 7.60997 79.4503 7.60997C79.0388 7.60997 78.761 7.32663 78.761 6.89638C78.761 6.27724 79.1828 5.66859 80.0469 5.16488C80.6745 4.79759 81.4872 4.5982 82.5468 4.5982C85.1495 4.5982 86.5898 5.89945 86.5898 8.26059V14.9138C86.5898 15.5434 86.2709 15.9107 85.7256 15.9107C85.201 15.9107 84.8923 15.5539 84.8718 14.9557V14.3366ZM79.9646 12.5526C79.9646 11.4298 80.7259 10.8001 82.053 10.8001H84.8203V12.0489C84.8203 13.4236 83.6578 14.473 82.1662 14.473C80.8494 14.473 79.9646 13.6965 79.9646 12.5526ZM91.4146 18.828V14.1792H91.5072C92.001 15.3335 93.1224 15.9841 94.6141 15.9841C97.1962 15.9841 98.8216 14.1687 98.8216 11.2619V9.30999C98.8216 6.42415 97.2168 4.5982 94.6758 4.5982C93.1224 4.5982 91.8673 5.33278 91.4249 6.52909H91.3323V5.65809C91.3323 5.02846 91.0237 4.67166 90.4888 4.67166C89.9538 4.67166 89.6452 5.03895 89.6452 5.67908V18.828C89.6452 19.4576 89.9744 19.8354 90.5299 19.8354C91.0854 19.8354 91.4146 19.4576 91.4146 18.828ZM91.4146 9.40443V11.3773C91.4146 13.1718 92.5874 14.41 94.2849 14.41C96.0337 14.41 97.0419 13.1193 97.0419 10.8841V9.68777C97.0419 7.44206 96.0337 6.1723 94.2334 6.1723C92.4948 6.1723 91.4146 7.38959 91.4146 9.40443ZM103.348 14.1792V18.828C103.348 19.4576 103.019 19.8354 102.463 19.8354C101.908 19.8354 101.579 19.4576 101.579 18.828V5.67908C101.579 5.03895 101.887 4.67166 102.422 4.67166C102.957 4.67166 103.266 5.02846 103.266 5.65809V6.52909H103.358C103.801 5.33278 105.056 4.5982 106.609 4.5982C109.15 4.5982 110.755 6.42415 110.755 9.30999V11.2619C110.755 14.1687 109.13 15.9841 106.548 15.9841C105.056 15.9841 103.935 15.3335 103.441 14.1792H103.348ZM103.348 11.3773V9.40443C103.348 7.38959 104.428 6.1723 106.167 6.1723C107.967 6.1723 108.975 7.44206 108.975 9.68777V10.8841C108.975 13.1193 107.967 14.41 106.218 14.41C104.521 14.41 103.348 13.1718 103.348 11.3773Z"
          fill="var(--icons)"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
  <h3
    class="pg-confirm__title"
  >
    Get Verified your email
    <svg
      class="pg-confirm__title__icon"
      fill="none"
      height="10"
      viewBox="0 0 10 10"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M3.33033 4.91061L0.367188 1.69265L1.83845 0.337891L4.68971 3.43434L7.54097 0.337891L9.01223 1.69265L6.04909 4.91061L9.0101 8.12624L7.53883 9.48101L4.68971 6.38688L1.84059 9.48101L0.369322 8.12624L3.33033 4.91061Z"
        fill="var(--system-red)"
        fill-rule="evenodd"
      />
    </svg>
  </h3>
  <div
    class="pg-confirm__content"
  >
    Something went wrong
  </div>
</div>
`;
